Panic when attempting to show and environment missing a spec.json

Using 0.3.0 image with an existing environment from ksonnet.
Steps to reproduce: 
- Perform `tk init -f` in the existing ksonnet project.
- Create an environment with a valid jsonnet file but missing the spec.json.
- Perform `tk show <environment>`
Error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x882fd8]

goroutine 1 [running]:
github.com/grafana/tanka/pkg/kubernetes.(*Kubernetes).Reconcile(0x0, 0xc0018803f0, 0x0, 0x0, 0x0, 0xc00048bce8, 0x5b8175, 0xc0000b8500, 0xc00048bcc0, 0xa6f1af)
        /drone/src/pkg/kubernetes/kubernetes.go:67 +0x118
main.showCmd.func1(0xc0000ce500, 0xc000164e90, 0x1, 0x1)
        /drone/src/cmd/tk/workflow.go:145 +0x122
github.com/spf13/cobra.(*Command).execute(0xc0000ce500, 0xc00001e1a0, 0x1, 0x1, 0xc0000ce500, 0xc00001e1a0)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000ce000, 0xc000164e20, 0xc00011e270, 0x0)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
        /drone/src/cmd/tk/main.go:106 +0x43f
```
Not a huge dealio, and the proper init workflow works fine it's something I encountered looking to switchover from ks to tk.

--- a/pkg/jpath/jpath.go
+++ b/pkg/jpath/jpath.go
@@ -9,10 +9,10 @@ import (
 
 var (
 	// ErrorNoRoot means no rootDir was found in the parents
-	ErrorNoRoot = errors.New("could not locate a jsonnetfile.json in the parent directories, which is required to identify the project root")
+	ErrorNoRoot = errors.New("could not locate a jsonnetfile.json in the parent directories, which is required to identify the project root. Refer to https://tanka.dev/directory-structure for more information")
 
 	// ErrorNoBase means no baseDir was found in the parents
-	ErrorNoBase = errors.New("could not locate a main.jsonnet in the parent directories, which is required as the entrypoint for the evaluation")
+	ErrorNoBase = errors.New("could not locate a main.jsonnet in the parent directories, which is required as the entrypoint for the evaluation. Refer to https://tanka.dev/directory-structure for more information")
 )
 
 // ErrorFileNotFound means that the searched file was not found

--- a/pkg/kubernetes/errors.go
+++ b/pkg/kubernetes/errors.go
@@ -9,3 +9,11 @@ type ErrorNotFound struct {
 func (e ErrorNotFound) Error() string {
 	return fmt.Sprintf(`error from server (NotFound): secrets "%s" not found`, e.resource)
 }
+
+type ErrorMissingConfig struct {
+	verb string
+}
+
+func (e ErrorMissingConfig) Error() string {
+	return fmt.Sprintf("%s requires additional configuration. Refer to https://tanka.dev/environments for that.", e.verb)
+}

--- a/pkg/kubernetes/kubernetes.go
+++ b/pkg/kubernetes/kubernetes.go
@@ -64,7 +64,9 @@ func (k *Kubernetes) Reconcile(raw map[string]interface{}, objectspecs ...string
 	}
 	for _, d := range docs {
 		m := objx.New(d)
-		m.Set("metadata.namespace", k.Spec.Namespace)
+		if k != nil {
+			m.Set("metadata.namespace", k.Spec.Namespace)
+		}
 		out = append(out, Manifest(m))
 	}
 
@@ -106,6 +108,10 @@ func (k *Kubernetes) Fmt(state []Manifest) (string, error) {
 
 // Apply receives a state object generated using `Reconcile()` and may apply it to the target system
 func (k *Kubernetes) Apply(state []Manifest, opts ApplyOpts) error {
+	if k == nil {
+		return ErrorMissingConfig{"apply"}
+	}
+
 	yaml, err := k.Fmt(state)
 	if err != nil {
 		return err
@@ -115,6 +121,9 @@ func (k *Kubernetes) Apply(state []Manifest, opts ApplyOpts) error {
 
 // Diff takes the desired state and returns the differences from the cluster
 func (k *Kubernetes) Diff(state []Manifest) (*string, error) {
+	if k == nil {
+		return nil, ErrorMissingConfig{"diff"}
+	}
 	yaml, err := k.Fmt(state)
 	if err != nil {
 		return nil, err

